lib/shakapacker/dev_server.rb

```
@@ -39,8 +39,29 @@ def https?
     end
   end
 
+  def server
+    server_value = fetch(:server)
+    server_type = server_value.is_a?(Hash) ? server_value[:type] : server_value
+
+    return server_type if ["http", "https"].include?(server_type)
+
+    return "http" if server_type.nil?
+
+    puts <<~MSG
+    WARNING:
+    `server: #{server_type}` is not a valid configuration in Shakapacker.
+    Falling back to default `server: http`.
+    MSG
+
+    "http"
+  rescue
```

this swallows all exceptions such as a typo

please always catch specific errors

---

Done